frameworks/hdfs/tests/config.py

```diff
@@ -56,15 +56,14 @@ def get_active_name_node(service_name):
     raise Exception("Failed to determine active name node")
 
 
+@retrying.retry(
+    wait_fixed=10000,
+    stop_max_delay=DEFAULT_HDFS_TIMEOUT*1000)
 def get_name_node_status(service_name, name_node):
-    def get_status():
-        rc, output = run_hdfs_command(service_name, "./bin/hdfs haadmin -getServiceState {}".format(name_node))
-        if not rc:
-            return rc
-
-        return output.strip()
-
-    return shakedown.wait_for(lambda: get_status(), timeout_seconds=DEFAULT_HDFS_TIMEOUT)
+    rc, output = run_hdfs_command(service_name, "./bin/hdfs haadmin -getServiceState {}".format(name_node))
+    if not rc:
+        return rc
```

Comment:
This is not clear that `rc` is returning a falsey value. Rather raise or return `None` or `False` explicitly.

Reply:
shakedown.wait_for works this way where it is fine with empty variables or False which I think has created a ton of confusion in the code. Here, we definitely need to add `retry_on_result=lambda res: not res`

Reply:
`rc` is an extremely confusing variable. Its not the return code, but rather its a boolean representing if the return code was 0 which behaves the opposite of how return codes would behave in this statement. not sure what this is actually supposed to be doing

Reply:
https://github.com/dcos/shakedown/blob/master/shakedown/dcos/command.py#L72

Reply:
Ok, so after talking to Nick, it seems the usage is correct, its just `rc` is very misleading and should be fixed in another PR